src/types/CodepointWidthDetector.cpp

```diff
@@ -310,13 +310,21 @@ namespace
     };
 }
 
+// Routine Description:
+// - Constructs an instance of the CodepointWidthDetector class
+CodepointWidthDetector::CodepointWidthDetector() noexcept :
+    _fallbackCache{},
+    _pfnFallbackMethod{}
+{
+}
+
 // Routine Description:
 // - returns the width type of codepoint by searching the map generated from the unicode spec
 // Arguments:
 // - glyph - the utf16 encoded codepoint to search for
 // Return Value:
 // - the width type of the codepoint
-CodepointWidth CodepointWidthDetector::GetWidth(const std::wstring_view glyph) const noexcept
+CodepointWidth CodepointWidthDetector::GetWidth(const std::wstring_view glyph) const

```

Comment:
should we make this one noexcept by catching and logging instead of letting it bubble out?

Reply:
I think a width error needs to be communicated to the caller...

Reply:
k